### PR TITLE
[Core] Removing 32 bit compilation support

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Removing 32 bit compilation support (@roigcarlo we decided to remove this in Delf)

**🆕 Changelog**

- [Removing 32 bit compilation support](https://github.com/KratosMultiphysics/Kratos/commit/ca93994d66e8bb1125ece5e9bddbeaac0a31fd58)
- [Revert 32 bit support change](https://github.com/KratosMultiphysics/Kratos/commit/cd4a413be69d2b890db8366441d7f14dd520829b)

### CODE REVIEW

**Comment by matekelemen** [comment]
rip rpi 🫡

**Comment by loumalouomega** [comment]
Rip 32bit pi, rpi 3 and 4 now are 64 by default with a significant increase
on performance

**Comment by matekelemen** [comment]
> rpi 3 and 4 now are 64 by default

Ah I didn't know that

File: kratos/includes/define.h [hunk 1]
```diff
@@ -40,12 +40,14 @@
...
    #else
      #define KRATOS_ENV32BIT
+     #error 32 bit system are not supported anymore. Please consider a 64 bits system
   #endif
 #else // It is POSIX (Linux, MacOSX, BSD...)
   #if defined(__x86_64__) || defined(__ppc64__) || defined(__aarch64__)
     #define KRATOS_ENV64BIT
   #else // This includes __arm__ and __x86__
     #define KRATOS_ENV32BIT
+     #error 32 bit system are not supported anymore. Please consider a 64 bits system
```
Comment (philbucher) [inline kratos/includes/define.h:50]: Can this be checked with CMake?
Seems like the better place to do this

Plus then we could get rid of these macros

Comment (loumalouomega) [reply kratos/includes/define.h:50]: We can...

Comment (loumalouomega) [reply kratos/includes/define.h:50]: Do you know how to check that in CMake?, I can't find an easy manner

Comment (roigcarlo) [reply kratos/includes/define.h:50]: I would just not check it, 32 bit users are scarse

Comment (loumalouomega) [reply kratos/includes/define.h:50]: But without this those scarce developers will be able to compile, and they will have an execution error not explained when using a variable and the hash induced and overflow

Comment (rubenzorrilla) [reply kratos/includes/define.h:50]: Maybe we can just inform about this in the wiki compilation guide.

Comment (loumalouomega) [reply kratos/includes/define.h:50]: You know that our main problem is that people don't read the documentation, I think the check is relevant and must be done

Comment (rubenzorrilla) [reply kratos/includes/define.h:50]: Well, I think that the how to compile is one of the very few that people always check... :confused:

Comment (loumalouomega) [reply kratos/includes/define.h:50]: But still we have the possibility that someone directly compiles without reading it. It is safer this way

Comment (roigcarlo) [reply kratos/includes/define.h:50]: Idk to be honest, I see @loumalouomega points, i don't like it but if you really want to leave it I have no problems with it. Maybe we can remove it later. What do you think?

**Comment by loumalouomega** [comment]
What is the status of this?

**Review by roigcarlo — APPROVED** [review]
I think we can approve. I still think we could just assume that 32bit does not exist but its ok.

**Pull request merged**